probe/probe.go

```diff
@@ -292,13 +322,339 @@ func probeNode(c *k8s.Client) {
 			}
 			fmt.Printf("\t Observability/Audit:")
 			kernelVersion := item.Status.NodeInfo.KernelVersion
-			check2 := checkNodeKernelHeaderPresent(c, item.Name, kernelVersion)
+			check2 := checkNodeKernelHeaderPresent(c, o, item.Name, kernelVersion)
 			checkAuditSupport(kernelVersion, check2)
-			lsm, _ := getNodeLsmSupport(c, item.Name)
+			lsm, _ := getNodeLsmSupport(c, o, item.Name)
 			checkLsmSupport(lsm)
 
 		}
 	} else {
 		fmt.Println("No kubernetes environment found")
 	}
 }
+
+type KubeArmorProbeData struct {
+	OSImage                 string
+	KernelVersion           string
+	KubeletVersion          string
+	ContainerRuntime        string
+	ActiveLSM               string
+	KernelHeaderPresent     bool
+	HostSecurity            bool
+	ContainerSecurity       bool
+	ContainerDefaultPosture tp.DefaultPosture
+	HostDefaultPosture      tp.DefaultPosture
+}
+
+func isKubeArmorRunning(c *k8s.Client, o Options) bool {
+	_, err := getKubeArmorDaemonset(c, o)
+	return err == nil
+
+}
+
+func getKubeArmorDaemonset(c *k8s.Client, o Options) (bool, error) {
+	var data [][]string
+	// KubeArmor DaemonSet
+	w, err := c.K8sClientset.AppsV1().DaemonSets(o.Namespace).Get(context.Background(), "kubearmor", metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+	color.Green("\nFound KubeArmor running in Kubernetes\n\n")
+	_, err = boldWhite.Printf("Daemonset :\n")
+	if err != nil {
+		color.Red(" Error while printing")
+	}
+	desired, ready, available := w.Status.DesiredNumberScheduled, w.Status.NumberReady, w.Status.NumberAvailable
+	if desired != ready && desired != available {
+		data = append(data, []string{" ", "kubearmor ", "Desired: " + strconv.Itoa(int(desired)), "Ready: " + strconv.Itoa(int(ready)), "Available: " + strconv.Itoa(int(available))})
+		return false, nil
+	}
+	data = append(data, []string{" ", "kubearmor ", "Desired: " + strconv.Itoa(int(desired)), "Ready: " + strconv.Itoa(int(ready)), "Available: " + strconv.Itoa(int(available))})
+	renderOutputInTableWithNoBorders(data)
+	return true, nil
+}
+func getKubeArmorDeployments(c *k8s.Client, o Options) {
+	var data [][]string
+	_, err := boldWhite.Printf("Deployments : \n")
+	if err != nil {
+		color.Red(" Error while printing")
+	}
+	kubearmorDeployments, err := c.K8sClientset.AppsV1().Deployments(o.Namespace).List(context.Background(), metav1.ListOptions{
+		LabelSelector: "kubearmor-app",
+	})
+
+	if err != nil {
+		return
+	}
+	if len(kubearmorDeployments.Items) > 0 {
+		for _, kubearmorDeploymentItem := range kubearmorDeployments.Items {
+			desired, ready, available := kubearmorDeploymentItem.Status.UpdatedReplicas, kubearmorDeploymentItem.Status.ReadyReplicas, kubearmorDeploymentItem.Status.AvailableReplicas
+			if desired != ready && desired != available {
+				data = append(data, []string{" ", kubearmorDeploymentItem.Name, "Desired: " + strconv.Itoa(int(desired)), "Ready: " + strconv.Itoa(int(ready)), "Available: " + strconv.Itoa(int(available))})
+				return
+			}
```

Comment:
What's the append and return doing here?

Reply:
oh, the return was an old thingy- i will remove it